nomad/job_endpoint.go

```diff
@@ -1298,19 +1298,72 @@ func (j *Job) GetJobVersions(args *structs.JobVersionsRequest,
 			// Setup the output
 			reply.Versions = out
 			if len(out) != 0 {
+
+				var compareVersionNumber uint64
+				var compareVersion *structs.Job
+				var compareSpecificVersion bool
+
+				if args.Diffs {
+					if args.DiffTagName != "" {
+						compareSpecificVersion = true
+						tagFound := false
+						for _, version := range out {
+							if version.TaggedVersion != nil && version.TaggedVersion.Name == args.DiffTagName {
+								tagFound = true
+								compareVersionNumber = version.Version
+								break
+							}
+						}
+						if !tagFound {
+							return fmt.Errorf("tag %q not found", args.DiffTagName)
+						}
+					} else if args.DiffVersion != nil {
+						compareSpecificVersion = true
+						compareVersionNumber = *args.DiffVersion
+					}
+				}
+
 				// Note: a previous assumption here was that the 0th job was the latest, and that we don't modify "old" versions.
 				// Adding version tags breaks this assumption (you can tag an old version, which should unblock /versions queries) so we now look for the highest ModifyIndex.
 				var maxModifyIndex uint64
 				for _, job := range out {
 					if job.ModifyIndex > maxModifyIndex {
 						maxModifyIndex = job.ModifyIndex
 					}
+					if compareSpecificVersion && job.Version == compareVersionNumber {
+						compareVersion = job
+					}
 				}
 				reply.Index = maxModifyIndex
+
+				if compareSpecificVersion && compareVersion == nil {
+					if args.DiffTagName != "" {
+						return fmt.Errorf("tag %q not found", args.DiffTagName)
+					}
+					return fmt.Errorf("version %d not found", *args.DiffVersion)
+				}
+
 				// Compute the diffs
+
+				// TODO:
+				// - tests
+				// - Remove job plan getting diffs and make it into a new command, `nomad job diff`
```

Comment:
Is this TODO for this PR or intended for another PR targeting the same working branch?